vault/provider.go

```diff
@@ -869,15 +869,14 @@ func ReadContextWrapper(f schema.ReadContextFunc) schema.ReadContextFunc {
 // wrapped schema.CreateContextFunc.
 func MinVersionCheckWrapper(f schema.CreateContextFunc, minVersion string) schema.CreateContextFunc {
 	return func(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-		client, e := provider.GetClient(d, meta)
+		mv, e := version.NewVersion(minVersion)
 		if e != nil {
 			return diag.FromErr(e)
 		}
 
-		featureEnabled, currentVersion, err := semver.GreaterThanOrEqual(ctx, client, minVersion)
-		if err != nil {
-			return diag.FromErr(err)
-		}
+		featureEnabled := provider.IsAPISupported(meta, mv)
+
+		currentVersion := meta.(*provider.ProviderMeta).GetVaultVersion().String()
```

Comment:
you can pass `currentVersion` directly to `diag.Errorf()` since it is a stringer  i.e. no need to call its `String()` method

Reply:
Nice! Didn't know that, thanks :)